packages/core/src/errorCollection.ts

```diff
@@ -27,26 +29,45 @@ export interface HttpContext {
   method: string
 }
 
-export enum ErrorSource {
+export interface AddedError {
+  startTime: number
+  error: unknown
+  context?: Context
+  source: ErrorSource
+}
+
+export enum InternalErrorSource {
   AGENT = 'agent',
   CONSOLE = 'console',
+  LOGGER = 'logger',
+}
+
+export enum ErrorSource {
```

Comment:
I intentionaly kept the `ErrorSource` name instead of having a `PublicErrorSource` enum, because this name is exported publicly (as a JS variable but also as a TS type).